tests/devices/test_default_qubit.py

```diff
@@ -870,6 +871,337 @@ def circuit():
         # an estimated variance an an analytically calculated one
         assert expval != 0.0
 
+    @pytest.mark.parametrize(
+        "qubits, operations, hamiltonian, expected_output",
+        [
+            (1, [], np.array([[1.0, 0.0], [0.0, 1.0]]), 1.0),
+            (
+                2,
+                [qml.PauliX(0), qml.PauliY(1)],
+                np.array(
+                    [
+                        [1.0, 0.0, 0.0, 0.0],
+                        [0.0, 1.0, 0.0, 0.0],
+                        [0.0, 0.0, 1.0, 0.0],
+                        [0.0, 0.0, 0.0, 1.0],
+                    ]
+                ),
+                -1.0,
+            ),
+            (
+                4,
+                [
+                    qml.PauliX(0),
+                    qml.PauliX(1),
+                    qml.DoubleExcitation(0.22350048065138242, wires=[0, 1, 2, 3]),
+                ],
+                np.array(
```

Comment:
Is it possible to describe them here directly as sparse matrices so they don't take so much space? Or with some other more compact method?